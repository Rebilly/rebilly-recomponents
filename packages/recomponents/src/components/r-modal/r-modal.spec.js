import {renderToString} from '@vue/server-test-utils';
import {shallowMount} from '@vue/test-utils';

import RModal from './r-modal.vue';

describe('r-modal.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RModal, {
            propsData: {
                size: 'large',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RModal, {
            propsData: {
                size: 'extra-large',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should emit all available event destroy', () => {
        const wrapper = shallowMount(RModal, {
            propsData: {
                size: 'fluid',
            },
            slots: {
                content: '<strong>default message</strong>',
            },
        });

        wrapper.vm.submit();
        wrapper.vm.close();
        wrapper.vm.enter();
        wrapper.destroy();
        expect(wrapper.emitted().submit).toBeTruthy();
        expect(wrapper.emitted().close).toBeTruthy();
        expect(wrapper.emitted().enter).toBeTruthy();
        expect(wrapper.emitted().leave).toBeTruthy();
    });
});
