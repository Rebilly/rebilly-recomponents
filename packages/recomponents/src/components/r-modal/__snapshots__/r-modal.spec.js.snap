// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-modal.vue should render Wrapper and match snapshot 1`] = `
<div tabindex="0" name="modal">
  <div class="r-modal-overlay">
    <r-loader-stub show="true" loading="true" class="r-modal-loader is-loading"></r-loader-stub>
  </div>
</div>
`;

exports[`r-modal.vue should render via SSR and match snapshot 1`] = `
<div tabindex="0" name="modal">
  <div class="r-modal-overlay">
    <div class="r-component r-loader r-modal-loader is-loading r-loader--loading">
      <div class="wrapper">
        <div class="mr-re"></div>
        <div class="mr-bil"></div>
        <div class="mr-ly"></div>
      </div>
    </div>
  </div>
</div>
`;
